instrumentation-api-semconv/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/http/HttpCommonAttributesGetter.java

```diff
@@ -58,9 +58,21 @@ default Long requestContentLengthUncompressed(REQUEST request, @Nullable RESPONS
    * <p>This is called from {@link Instrumenter#end(Context, Object, Object, Throwable)}, only when
    * {@code response} is non-{@code null}.
    */
+  // TODO: deprecate this method and use the new one everywhere
   @Nullable
   Integer statusCode(REQUEST request, RESPONSE response);
 
+  /**
+   * Extracts the {@code http.status_code} span attribute.
+   *
+   * <p>This is called from {@link Instrumenter#end(Context, Object, Object, Throwable)}, only when
+   * {@code response} is non-{@code null}.
+   */
+  @Nullable
+  default Integer statusCode(REQUEST request, RESPONSE response, @Nullable Throwable error) {
+    return statusCode(request, response);
+  }
```

Comment:
I wanted the Spring/Servlet status code to be properly reflected in the span status, so I added this method. I'll deprecate the previous one in the next PR.